src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { getActiveTodos, getCompletedTodos, getTodos } from '../../api/todos';
+
+interface FooterProps {
+  visibleTodos: Todo[];
+  selectedFilter: string;
+  setSelectedFilter: React.Dispatch<React.SetStateAction<string>>;
+  setVisibleTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+}
+
+export const Footer: React.FC<FooterProps> = ({
+  visibleTodos,
+  selectedFilter,
+  setSelectedFilter,
+  setVisibleTodos,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {visibleTodos.filter(todo => !todo.completed).length} items left
+      </span>
+
+      {/* Active link should have the 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={classNames('filter__link', {
+            selected: selectedFilter === 'all',
+          })}
+          data-cy="FilterLinkAll"
+          onClick={() => {
+            if (selectedFilter === 'all') {
+              return;
+            }
+
+            setSelectedFilter('all');
+            getTodos().then(allTodos => setVisibleTodos(allTodos));
```

Comment:
Use `Object.values(your created enum)` and render these options with `map()` method